SimCalorimetry/EcalSimAlgos/interface/EcalCoder_Ph2.h

```diff
@@ -0,0 +1,83 @@
+#ifndef SimCalorimetry_EcalSimAlgos_EcalCoder_Ph2_h
+#define SimCalorimetry_EcalSimAlgos_EcalCoder_Ph2_h
+
+#include "CalibFormats/CaloObjects/interface/CaloTSamples_Ph2.h"
+#include "CondFormats/EcalObjects/interface/EcalLiteDTUPedestals.h"
+#include "CondFormats/EcalObjects/interface/EcalIntercalibConstantsMC.h"
+#include "CondFormats/EcalObjects/interface/EcalCATIAGainRatios.h"
+#include "SimCalorimetry/EcalSimAlgos/interface/EcalCorrelatedNoiseMatrix_Ph2.h"
+#include "CondFormats/EcalObjects/interface/EcalConstants.h"
+
+template <typename M>
+class CorrelatedNoisifier;
+class EcalLiteDTUSample;
+class EcalDataFrame_Ph2;
+class DetId;
+class EcalLiteDTUPed;
+
+#include <vector>
+
+namespace CLHEP {
+  class HepRandomEngine;
+}
+
+class EcalCoder_Ph2 {
+public:
+  typedef CaloTSamples_Ph2<float, ecalPh2::sampleSize> EcalSamples;
+
+  //typedef CaloTSamples_Ph2<float,10> EcalSamples ;
+
+  typedef CorrelatedNoisifier<EcalCorrMatrix_Ph2> Noisifier;
+
+  enum {
```

Comment:
do these need to be enums, or could the `ecalPh2` constant values just be used directly?